### PR TITLE
SCUMM: DiMUSE: Fix bug #13080

### PR DESCRIPTION
**Pull request by AndywinXp**
For [this bug](https://bugs.scummvm.org/ticket/13080) to be fixed, I needed to avoid setting an arbitrary `_maxQueuedStreams` value in `IMuseDigital::tracksCallback()`, and instead do a proper calculation based on the audio buffer size.

To do that I needed the audio backend to pass the buffer size along to the mixer (with which DiMUSE can communicate). This has been done for all the ports/platform I could be able to test. For the others I left a fallback value of 4 for the maximum number of queued streams (until someone can test all the other platforms).

The changes in this PR have also been tested with various manually inserted values for `output_rate` (22050, 44100, 48000) and `audio_buffer_size` (from 512 up to 8192) in the `scummvm.ini` config files, and I can happily report that everything works as it should both in DiMUSE and in a bunch of other games and engines which I didn't touch (just for good measure).

Unfortunately Android works in its own ways, and reports a much higher latency than the real one, resulting in a very high `_maxQueuedStreams`  value (and therefore high latency). I have solved this by getting the Android backend to record the actual audio buffer size, which I retrieved from the AudioTrack object _after_ its creation. But this fix will be put in a separate PR, just for safety.

### CODE REVIEW

File: audio/mixer_intern.h [hunk 1]
```diff
@@ -81,7 +82,7 @@ class MixerImpl : public Mixer {
 
 public:
 
-	MixerImpl(uint sampleRate);
+	MixerImpl(uint sampleRate, uint outBufSize = -1);
```
Comment (aquadran) [inline audio/mixer_intern.h]: assign -1 to uint is good idea. 0 is not enough?

**Review by aquadran — CHANGES_REQUESTED** [review]

File: audio/mixer_intern.h (hunk 1 above)
Comment (AndywinXp) [reply audio/mixer_intern.h]: Whoops, it should have been an int, not a unit.
But I suppose you are right, I can't think of a reason why -1 might be better than 0 as a special value.

**Comment by aquadran** [comment]
can you squash last and "AUDIO: Set _outBufSize to signed int" commits?

**Comment by AndywinXp** [comment]
Thanks, done

File: audio/mixer_intern.h [hunk 2]
```diff
@@ -82,7 +82,7 @@ class MixerImpl : public Mixer {
 
 public:
 
-	MixerImpl(uint sampleRate, uint outBufSize = -1);
+	MixerImpl(uint sampleRate, uint outBufSize = 0);
```
Comment (aquadran) [inline audio/mixer_intern.h:85]: can you squash this change to earlier commit which introduce -1

Comment (AndywinXp) [reply audio/mixer_intern.h:85]: Sure, I'll look into it and do it

Comment (AndywinXp) [reply audio/mixer_intern.h:85]: There you go, should be much more cleaner than before

**Review by aquadran — APPROVED** [review]

**Pull request merged**